File: src/com/googlecode/utterlyidle/ApplicationServlet.scala
package org.webfabric.rest

import javax.servlet.http.{HttpServletResponse, HttpServletRequest, HttpServlet}
import javax.servlet.ServletConfig

class ApplicationServlet extends HttpServlet{
  var application:RequestHandler = null

  override def init(config: ServletConfig) = {
    application = config.getServletContext.getAttribute(classOf[Application].getCanonicalName).asInstanceOf[RequestHandler]
  }

  override def service(req: HttpServletRequest, resp: HttpServletResponse) = {
    application.handle(Request(req), Response(resp))
  }
}

File: src/com/googlecode/utterlyidle/ApplicationServlet.java
package com.googlecode.utterlyidle;

import javax.servlet.http.*;
import javax.servlet.ServletConfig;

import static com.googlecode.utterlyidle.Request.request;
import static com.googlecode.utterlyidle.Response.response;

public class ApplicationServlet extends HttpServlet{
  RequestHandler application = null;

  @Override
  public void init(ServletConfig config) {
    application = (RequestHandler) config.getServletContext().getAttribute(Application.class.getCanonicalName());
  }

  @Override
  public void service(HttpServletRequest req, HttpServletResponse resp){
    application.handle(request(req), response(resp));
  }
}